dds/DCPS/Marked_Default_Qos.cpp

```diff
@@ -19,9 +19,13 @@ int const INVALID_ENUM_VALUE = 999;
 DDS::DomainParticipantQos
 Marked_Default_Qos::marked_default_DomainParticipantQos()
 {
-  DDS::DomainParticipantQos qos;
-  DDS::DomainParticipantFactory_var factory = TheParticipantFactory;
-  factory->get_default_participant_qos(qos);
+  DDS::DomainParticipantQos qos = TheServiceParticipant->initial_DomainParticipantQos();
+  qos.user_data.value.length(5);
```

Comment:
The "marked" value needs to be a value that's not possible for a user to select.  For DomainParticipant that would have to be autoenable_
created_entities not being true or false (i.e. treating the bool as a char).